Standards/scs-0303-v1-standard-roles.md

```diff
@@ -0,0 +1,193 @@
+
+### Options considered
+
+#### Using the current OpenStack default roles as-is
+
+At the time of writing this standard, OpenStack is in the midst of updating and unifying the default roles and their permission set in an ongoing RBAC rework[^2].
+This will update the set of default roles and their permission sets.
+
+Currently, Barbican still uses a separate set of default roles[^3] that is distinct from Keystone and the rest of the services.
+However, this will be replaced by the unified roles with the RBAC rework.
+
+Due to this, the current state of the default roles and permissions in OpenStack can be considered depracated and will change as soon as the RBAC rework is concluded.
+
+As a result, this option is not a good choice for standardization in SCS, because it would be based on soon-to-be outdated OpenStack defaults.
+
+[^2]: [OpenStack Technical Committee Governance Documents: Consistent and Secure Default RBAC](https://governance.openstack.org/tc/goals/selected/consistent-and-secure-rbac.html)
+
+[^3]: [OpenStack Barbican Documentation: Access Control - Default Policy](https://docs.openstack.org/barbican/2023.2/admin/access_control.html#default-policy)
+
+#### Fully using the RBAC rework of OpenStack as a basis
+
+Instead of focusing on the current defaults of OpenStack at the time of writing this standard, the upcoming RBAC rework[^1] could serve as a basis for a unified role and permission set in SCS.
+
+For this option, SCS would adopt the upcoming changes to OpenStack's default roles in advance and only incorporate minimal adjustments where necesary.
+
+The risk associated to this approach is related to the parts of the RBAC rework that are not finally implemented yet.
+For instance, this concerns the "manager" role that OpenStack aims to establish on a project-scope level.
+In case this standard would prematurely adopt role and policy configuration that get changed until the RBAC rework concludes, it would introduce unnecessary divergence and burden CSPs with another transition later on.
+
+#### Using only the stable parts of the OpenStack RBAC rework as a basis
+
+Instead of prematurely adopting all proposed changes of the upcoming RBAC rework in OpenStack[^1], this standard could limit its adoption scope to changes of the rework that have already been finalized in OpenStack at the time of writing this standard.
+This way, no role or policy definitions would be incorporated that pose the risk of still receiving major changes from upstream.
+
+Once the RBAC rework concludes, anything not already included in this standard could be added to it in a later version/revision of the standard, seamlessly aligning with upstream later on.
+
+## Open questions
+
+### Limited scope of OpenStack services covered by this standard
+
+Currently, SCS does not enforce a list of OpenStack components/services it covers and/or supports exactly.
```

Comment:
How do we want to address this?

Reply:
See the corresponding action item here: https://github.com/SovereignCloudStack/minutes/blob/main/iam/20231018.md?plain=1#L65-L72

Quote:
> - = predefine policy.yaml files for all services that have an API
>     - Q: does SCS include a fixed set of services?
>         - AI @garloff -> @berendt, @fkr: Put all services in a number of buckets:
>             1. Standard services (expected to be available with each SCS deployment)
>             2. Optional but fully supported
>             3. Optional, no guarantees (Tech Preview ...)
>             4. Unavailable